akka-stream/src/main/scala/akka/stream/impl/Timers.scala

```diff
@@ -138,13 +138,21 @@ object Timers {
 
   }
 
-  final class IdleTimeoutBidi[I, O](val timeout: FiniteDuration) extends GraphStage[BidiShape[I, I, O, O]] {
+  /**
+   * @param timeoutToTermination allows customising the behaviour once a timeout triggers.
+   *                             Returning None causes completion, and Some(exception) causes failure of the stage with that exception.
+   */
+  final class IdleTimeoutBidi[I, O](val timeout: FiniteDuration, timeoutToTermination: () ⇒ Option[Throwable]) extends GraphStage[BidiShape[I, I, O, O]] {
```

Comment:
Pretty sure that would log it as error then – since it would not be failStage but a throw inside onUpstreamFailure.